### DISCUSSION TITLE
Nova3D Whale 3 CWS files not working with UVT

### DISCUSSION THREAD

**Discussion by Balerion84** [sn4k3/UVtools#881] [Q&A]
Hey, so currently, CWS files, that nova3d is using for their whale 3 series printers, can't be edited with uv tools. If one tries to apply wait times via UV Tools for example, the saved file won't print. Plate will go down, wait for the set amount of time, then displays Error (just a general error message) and the print will stop. Apparently UVT messes up the code somehow, but I don't have the knowledge required to understand what :) 

Is it possible to have this fixed so that UV Tools can be used with the cws files?

**Comment by sn4k3** [comment]
Attach here an unmodified file for your printer.
CWS is universal format, if they using different gcode or images than original CWS odds are no good

**Reply by Balerion84** [reply]
so this is an unmodified cws file from tango slicer for the Nova3D Whale3 SE printer  (I just renamed the file extension to .zip from .cws as github didn't like the cws extension)
[Box.zip](https://github.com/user-attachments/files/15520834/Box.zip)

**Reply by sn4k3** [reply]
I don't see any reason for that to happen, everything looks normal. What error message does it show?

Meanwhile try this file: [Box_01h56m_1ml_copy.zip](https://github.com/user-attachments/files/15522304/Box_01h56m_1ml_copy.zip)

**Reply by Balerion84** [reply]
That's part of the problem, I don't know why it doesn't work, the error message just says "printing error" 
it stops before the exposure starts on the first layer
the plate goes down, print starts, time starts, printer appears to wait for a bit longer than then the 60s and then it stops with error message without any exposure

I'll try the file you sent and report back

**Reply by Balerion84** [reply]
Looks like that file worked, no error this time (preview picture on the printer was corrupted but that's not important), it actually printed before I stopped it to check. What changes did you make?

**Reply by sn4k3** [reply]
I just moved the  `;<Slice>` line to the begin of each layer. Should not impact but they can have any verification that abort print if not the first line after a home, idk
